packages/data-context/src/codegen/spec-options.ts

```diff
@@ -18,18 +21,104 @@ interface CodeGenOptions {
 //   Button.foo.js  -> Button.foo-copy-1.js
 export const expectedSpecExtensions = ['.cy', '.spec', '.test', '-spec', '-test', '_spec']
 
+type ComponentExtension = '.cy.js' | '.cy.ts' | '.cy.jsx' | '.cy.tsx'
+type TemplateKey = 'e2e' | 'componentEmpty' | 'vue2Component' | 'vue3Component'
 export class SpecOptions {
   private parsedPath: ParsedPath;
+  private parsedErroredCodegenCandidate?: ParsedPath
+
+  private async getFrontendFramework () {
+    const detected = await detectFramework(this.ctx.currentProject || '')
```

Comment:
I think we should use `cypress.config` and get this from `component.devServer` - no need to detect, they should have told us what they are using.

Reply:
I agree, but in this case I'm going to need to do something similar to what `detectFramework` does (look through satisfied dependencies) anyway because I need to differentiate between Vue 2 and Vue 3 (because they use different file templates). The config uses `framework: 'vue-cli'` for both Vue 2 and 3.

Reply:
Hmm but you can use this for both v2 and v3:

```html
<script>
export default {
}
</script>

<template>
</template>
```

And it's valid. I wonder if we can really assume everyone using `script setup`? This is my original issue with not making this customizable - unless you get lucky, you'll need to change the generated content - which means the feature is probably not super useful, until you can customize the content.

Until then, can we just pick a template that works for both v2 and v3 (something simple like what's suggested above?)

Reply:
@lmiller1990 I'm not sure what you mean. The templates that we have are for generating specs for vue 2 and vue 3 components with the only difference being `props: {}` vs `propData: {}` corresponding to the different Vue Test Util versions I believe. I don't think those rely on everyone using `script setup` syntax in their Vue components.

I agree that we should allow users to customize or provide their own templates in the future though

https://github.com/cypress-io/cypress/blob/99ce5c26522aebed1115cdf439730f5d412c296e/packages/data-context/src/codegen/templates/vue2-component/vue2-component.ejs